pymongo/client_session.py

```diff
@@ -379,7 +379,7 @@ def commit_transaction(self):
         .. versionadded:: 3.7
         """
         self._check_ended()
-
+        retry = False
```

Comment:
Let's add a comment to clarify what this flag is trying to track. AFAICT this is present to track explicit retries (application called `commit_transaction` twice).